core/src/main/java/be/orbinson/aem/dictionarytranslator/servlets/action/ImportDictionaryServlet.java

```diff
@@ -149,16 +150,6 @@ private void processCsvRecord(Resource dictionaryResource, List<String> language
             String translation = csvRecord.get(language);
             translations.get(index).add(translation);
 
-            createOrUpdateMessageEntryResource(dictionaryResource, resourceResolver, language, key, translation);
-        }
-    }
-
-    private void createOrUpdateMessageEntryResource(Resource dictionaryResource, ResourceResolver resourceResolver, String language, String key, String translation) throws PersistenceException, RepositoryException {
-        Resource languageResource = dictionaryService.getLanguageResource(dictionaryResource, language);
-        Resource messageEntryResource = dictionaryService.getMessageEntryResource(languageResource, key);
-        if (messageEntryResource == null) {
-            dictionaryService.createMessageEntry(resourceResolver, dictionaryResource, language, key, translation);
-        } else {
             dictionaryService.updateMessageEntry(resourceResolver, dictionaryResource, language, key, translation);
```

Comment:
what if the dictionary is a JSON-based dictionary?

Reply:
Good point, currently the logic seems to be merge, i.e. existing keys are not touched. Therefore I would rather bail out with an exception in case this has a different format.